### DISCUSSION TITLE
Using a package that's on main but not yet released?

### DISCUSSION THREAD

**Discussion by akshayka** [pyodide/pyodide#4737] [Q&A]
Hi Pyodide team!

PR #4587 added support for CVXPY and Clarabel, but it looks like these packages weren't included in 0.25.1.

Is there anyway I can use them with Pyodide today, or soon?

For context, I'm helping teach a course about CVXPY that starts early June, and I'd love to have our students use CVXPY through Pyodide.

**Comment by ryanking13** [comment]
We will release 0.26.0 in a few weeks, so you will be able to use it when we make a release.

Otherwise, I think there are two more options you can try:

1. We upload the tip-of-tree version of Pyodide on every commit to the CDN: https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.js, so you can use it.

```html
<!doctype html>
<html>
  <head>
      <script src="https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.js"></script>
  </head>
</html>
```

2. If you don't want to rely on the unstable tip-of-tree Pyodide, you can download the cvxpy and its dependencies from the CDN

  - https://cdn.jsdelivr.net/pyodide/dev/full/cvxpy_base-1.5.0-py3-none-any.whl
  - https://cdn.jsdelivr.net/pyodide/dev/full/clarabel-0.7.1-cp37-abi3-emscripten_3_1_52_wasm32.whl

and host it somewhere, and use `pyodide.loadPackage("<wheel_url>")` to load those packages.

**Reply by akshayka** [reply]
Thank you for the detailed write-up. This is enough to unblock us; thanks!

**Reply by akshayka** [reply]
I tried the second suggestion and attempted to micropip CVXPY and its dependencies from the CDN. I received the following error:

```
This cell raised an exception: ValueError('Wheel was built with Emscripten v3.1.52 but Pyodide was built with Emscripten v3.1.46')
```

Am I correct that my options are to either rebuild the wheel myself, or try using the tip-of-tree Pyodide version?

**Reply by hoodmane** [reply]
Yeah, you could use for instance pyodide 0.25.0a4 which has cvxpy or build it yourself.

**Reply by akshayka** [reply]
Perfect, thank you! Got it working.
